src/gravel/controllers/nodes/local.py

```diff
@@ -0,0 +1,185 @@
+# project aquarium's backend
+# Copyright (C) 2021 SUSE, LLC.
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+
+import asyncio
+from enum import Enum
+from typing import List
+
+import psutil
+from pydantic import BaseModel, Field
+
+# Minimum requirements for individual host validation:
+# NOTE(jhesketh): These are obviously hardcoded for the moment, but may be
+#                 better moved into some kind of support matrix (eg feature
+#                 vs hardware etc).
+#                 These are also currently somewhat arbitrary and will need
+#                 tuning.
+AQUARIUM_MIN_CPU = 2
```

Comment:
I think it would be preferable to use bytes here instead of GB, and let the frontend do the conversion in a consistent manner.

Reply:
I assume this comment was meant for memory and maybe disk, but good idea.